### PR TITLE
Feature/boas 1399 pins data model

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* feat(api/applications): install pins-data-model into API
* feat(api/applications): change migrator service to use pins-data-model
* feat(api/applications): use pins-data-model in schema-test-utils
* feat(api/applications): remove message schema files
These now come from the pins-data-model repository.

## Issue ticket number and link

[BOAS-1399](https://pins-ds.atlassian.net/browse/BOAS-1399)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[BOAS-1399]: https://pins-ds.atlassian.net/browse/BOAS-1399?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/api/src/server/migration/migrator.service.js [hunk 1]
```diff
@@ -38,16 +38,12 @@ export const getMigratorForModel = (modelType) => {
 };
 
 // the mappings are lazily initialised to improve test performance
-const initializeMapping = () => {
-	const schemas = readdirSync('./src/message-schemas/events')
-		.filter((file) => file.endsWith('.schema.json'))
-		.map((file) => {
-			return JSON.parse(readFileSync(`./src/message-schemas/events/${file}`).toString());
-		});
+const initializeMapping = async () => {
```
Comment (beejjacobs) [inline apps/api/src/server/migration/migrator.service.js:41]: Do we need a change somewhere to await this call? Wasn't async before.

File: apps/api/package.json [hunk 2]
```diff
@@ -72,6 +72,7 @@
     "pino": "^8.1.0",
     "pino-http": "^8.1.0",
     "pino-pretty": "^8.1.0",
+    "pins-data-model": "github:Planning-Inspectorate/data-model#draft-01",
```
Comment (beejjacobs) [inline apps/api/package.json:75]: I'd expect a package-lock change along with this, also draft-02 is latest now

**Review by beejjacobs — COMMENTED** [review]
Could of minor points - but looks good!

File: apps/api/src/server/migration/migrator.service.js (hunk 1 above)
Comment (HadrianHughesKC) [reply apps/api/src/server/migration/migrator.service.js:41]: Good spot, that's fixed.

File: apps/api/package.json (hunk 2 above)
Comment (HadrianHughesKC) [reply apps/api/package.json]: Updated to draft-02 and the package-lock change is there now 👍

**Review by beejjacobs — APPROVED** [review]
LGTM!

**Pull request merged**